tests/integration/conftest.py

```diff
@@ -82,7 +109,20 @@ async def app(ops_test: OpsTest, app_name: str, app_config: Dict[str, str], pyte
     await asyncio.gather(
         ops_test.model.add_relation(app_name, "postgresql-k8s:db-admin"),
         ops_test.model.add_relation(app_name, "redis-k8s"),
+        ops_test.model.add_relation(app_name, "nginx-ingress-integrator"),
     )
     await ops_test.model.wait_for_idle(status="active")
 
     yield application
+
+
+@pytest_asyncio.fixture(scope="module")
+async def setup_saml_config(ops_test: OpsTest, app: Application):
+    """Set SAML related charm config to enable SAML authentication."""
+    assert ops_test.model
+    discourse_app = ops_test.model.applications[app.name]
+    await discourse_app.set_config(
+        {"saml_target_url": "https://login.staging.ubuntu.com/+saml", "force_https": "true"}
+    )
+    yield
+    await discourse_app.set_config({"saml_target_url": "", "force_https": "false"})
```

Comment:
Should this revert back to the starting configuration rather than hardcode? E.g., capture the configuration at the start of the fixture, set the configuration and then revert back to the starting configuration?

Reply:
Updated, thanks!